multichain-testing/test/stake-ica.test.ts

```diff
@@ -0,0 +1,220 @@
+import anyTest from '@endo/ses-ava/prepare-endo.js';
+import type { TestFn } from 'ava';
+import { commonSetup, SetupContextWithWallets } from './support.js';
+import { makeDoOffer } from '../tools/e2e-tools.js';
+import { makeQueryClient } from '../tools/query.js';
+
+const test = anyTest as TestFn<SetupContextWithWallets>;
+
+const accounts = ['user1', 'user2'];
+
+test.before(async t => {
+  const { deleteTestKeys, setupTestKeys, ...rest } = await commonSetup(t);
+  // deleteTestKeys().catch();
+  const wallets = await setupTestKeys(accounts);
+  t.context = { ...rest, wallets, deleteTestKeys };
+});
+
+test.after(async t => {
+  const { deleteTestKeys } = t.context;
+  deleteTestKeys().catch();
+});
+
+interface Scenario {
+  chain: string;
+  chainId: string;
+  contractName: string;
+  denom: string;
+  expectedAddressPrefix: string;
+  /** path to contract proposal builder. must create a plan.json */
+  builder: string;
+  wallet: string;
+}
+
+const stakeScenario = test.macro(async (t, scenario: Scenario) => {
+  const {
+    wallets,
+    provisionSmartWallet,
+    makeQueryTool,
+    retryUntilCondition,
+    useChain,
+    deployBuilder,
+  } = t.context;
+
+  t.log('bundle and install contract');
+  await deployBuilder(scenario.builder);
+
+  const wdUser1 = await provisionSmartWallet(wallets[scenario.wallet], {
+    BLD: 100n,
+    IST: 100n,
+  });
+  t.log(`provisioning agoric smart wallet for ${wallets[scenario.wallet]}`);
+
+  const vstorageClient = makeQueryTool();
+
+  const doOffer = makeDoOffer(wdUser1);
+  t.log(`${scenario.contractName} makeAccountInvitationMaker offer`);
+  const makeAccountofferId = `makeAccount-${Date.now()}`;
+
+  // FIXME we get payouts but not an offer result; it times out
```

Comment:
_question_

Did you plan to address this in this PR or a separate PR?

Reply:
This is the main blocker on the PR currently. Ideally, we can square this away here and read offer results from the Notifier / `@agoric/casting` instead of polling vstorage.

However, since we are still able to read offer results in vstorage, there is an opportunity to fix forward if we wish to land this faster.

Reply:
Created https://github.com/Agoric/agoric-sdk/issues/9643 so we can tackle this in a separate PR.